kloppy/domain/services/transformers/__init__.py

```diff
@@ -74,11 +92,45 @@ def __needs_flip(
         return flip
 
     def transform_frame(self, frame: Frame) -> Frame:
-        flip = self.__needs_flip(
+
+        # Change coordinate system
+        if self._to_coordinate_system is not None:
```

Comment:
I really like you did this in 3 separate steps. I might come with a performance penalty but can be solved by saving the transformed output before continue processing, so you don't have to do the transformation everytime